server/src/main/java/org/opensearch/search/backpressure/trackers/NodeDuressTrackers.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.backpressure.trackers;
+
+import org.opensearch.common.util.Streak;
+
+import java.util.function.BooleanSupplier;
+import java.util.function.IntSupplier;
+
+/**
+ * NodeDuressTrackers is used to check if the node is in duress based on various resources.
+ *
+ * @opensearch.internal
+ */
+public class NodeDuressTrackers {
+
+    private final NodeDuressTracker heapDuressTracker;
+    private final NodeDuressTracker cpuDuressTracker;
+
+    public NodeDuressTrackers(final NodeDuressTracker heapDuressTracker, final NodeDuressTracker cpuDuressTracker) {
+        this.heapDuressTracker = heapDuressTracker;
+        this.cpuDuressTracker = cpuDuressTracker;
+    }
```

Comment:
NodeDuressTrackers can be generic by maintaining map of duress trackers against each resource. The resource type can be enum returned by the tracker itself, and constructor can throw error for duplicate duress tracker

Reply:
And, isHeapInDuress methods can change to `isResourceInDuress(ResourceType parameter)`

Reply:
I will refactor this